src/main/native/cpp/analysis/FilteringUtils.cpp

```diff
@@ -91,3 +138,168 @@ units::second_t sysid::GetMeanTimeDelta(const Storage& data) {
 
   return std::accumulate(dts.begin(), dts.end(), 0_s) / dts.size();
 }
+
+void sysid::TrimQuasistaticData(std::vector<PreparedData>* data,
+                                double motionThreshold) {
+  data->erase(std::remove_if(data->begin(), data->end(),
+                             [motionThreshold](const auto& pt) {
+                               return std::abs(pt.voltage) <= 0 ||
+                                      std::abs(pt.velocity) < motionThreshold;
+                             }),
+              data->end());
+}
+
+void sysid::ApplyMedianFilter(std::vector<PreparedData>* data, int window) {
+  size_t step = window / 2;
+  frc::MedianFilter<double> medianFilter(window);
+
+  // Load the median filter with the first value, step times for accurate
+  // behaviour at the start.
+
```

Comment:
```suggestion
  // Load the median filter with the first value "step" number of times for
  // accurate initial behavior.
```
Also remove the blank line before the for loop since this comment refers to that block of code specifically.